### PR TITLE
Schema docs generation extended types

### PR DESCRIPTION
**Pull request by dalepotter**
Adds support for documentation of `complexType` elements (that extend other `complexType` elements), which are added in the v2.03 schemas.

### CODE REVIEW

File: gen.py [hunk 1]
```diff
@@ -165,38 +186,58 @@ def __init__(self, schema, lang):
...
         if ref_element is not None:
             xsd_docuementation = ref_element.find(".//xsd:documentation", namespaces=namespaces)
             if xsd_docuementation is not None:
                 return ref_element.find(".//xsd:documentation", namespaces=namespaces).text
         return element.find(".//xsd:documentation", namespaces=namespaces).text
 
 
-    def output_docs(self, element_name, path, element=None, minOccurs='', maxOccurs='', ref_element=None):
+    def output_docs(self, element_name, path, element=None, minOccurs='', maxOccurs='', ref_element=None, type_element=None):
         """
```
Comment (allthatilk) [inline gen.py]: Description could use bringing in mine with rest of linting style.

File: gen.py [hunk 2]
```diff
@@ -326,15 +368,22 @@ def output_overview_page(self, standard, page, reference_pages):
             ).encode('utf8'))
 
 
-
     def element_loop(self, element, path):
+        """Find child elements for a given input element.
+
+        Args:
+            element (lxml.etree._Element): The base element to find child elements for.
+            path (str): Unused.
```
Comment (allthatilk) [inline gen.py:375]: If it's not used can it be removed?

**Review by allthatilk — CHANGES_REQUESTED** [review]
I mean, these changes aren't that important so if they get ignored no big.

File: gen.py (hunk 2 above)
Comment (dalepotter) [reply gen.py:375]: With there being no tests for this function, I'm not comfortable removing parameters!

File: gen.py (hunk 1 above)
Comment (dalepotter) [reply gen.py]: Fixed in 963b481.

**Review by allthatilk — APPROVED** [review]

**Pull request merged**